### PR TITLE
tests: fs_api: move to new ztest API

### PR DESCRIPTION
**Pull request by mengxianglinx**
all test cases in tests/subsys/fs/fs_api are moved to new ztest API

Signed-off-by: Meng xianglin <xianglinx.meng@intel.com>

### CODE REVIEW

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 1]
```diff
@@ -1031,3 +1034,53 @@ void test_file_unlink(void)
...
 	TC_PRINT("File (%s) deleted successfully!\n", TEST_FILE_RN);
 }
+
+static void *fs_api_setup(void)
+{
+	fs_register(TEST_FS_1, &temp_fs);
+	fs_mount(&test_fs_mnt_1);
+	fs_register(TEST_FS_2, &null_fs);
+	null_fs.mount = temp_fs.mount;
+	null_fs.unmount = temp_fs.unmount;
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1044]: Shouldn't these precede `fs_register`?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 2]
```diff
@@ -1031,3 +1034,53 @@ void test_file_unlink(void)
...
+	fs_mount(&test_fs_mnt_no_op);
+	return NULL;
+}
+
+static void fs_api_teardown(void *fixtrue)
+{
+	fs_unmount(&test_fs_mnt_no_op);
+	fs_unregister(TEST_FS_2, &null_fs);
+	null_fs.mount = null_fs.unmount = NULL;
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1053]: Shouldn't this line precede the `fs_register`?

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 3]
```diff
@@ -191,6 +191,9 @@ void test_unmount(void)
 	TC_PRINT("unmount a file system has no unmount functionality\n");
 	ret = fs_unmount(&test_fs_mnt_no_op);
 	zassert_not_equal(ret, 0, "Unmount a fs has no unmount functionality");
+	/* assign a unmount interface to test_fs_mnt_no_op to unmount it */
+	null_fs.unmount = temp_fs.unmount;
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:195]: Comment claims assignment to `test_fs_mnt_no_op` but assignment is done to `null_fs`

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c [hunk 4]
```diff
@@ -191,6 +191,9 @@ void test_unmount(void)
...
+	ret = fs_unmount(&test_fs_mnt_no_op);
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs_dir_file.c:196]: `ret` is not checked.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 1 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1046]: Yes, thank you!

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 2 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c]: Updated.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 3 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c]: updated.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 4 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:197]: updated.

**Review by yinfangw — DISMISSED** [review]

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 4 above)
Comment (enjiamai) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:197]: Comment addressed.

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 1 above)
Comment (enjiamai) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:1046]: Comment addressed.

**Review by enjiamai — DISMISSED** [review]

File: tests/subsys/fs/fs_api/src/test_fs_dir_file.c (hunk 4 above)
Comment (mengxianglinx) [reply tests/subsys/fs/fs_api/src/test_fs_dir_file.c:197]: @de-nordic code is updated, can help to review again? thank you!

**Review by de-nordic — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Pull request merged**